pythonx/UltiSnips/snippet/definition/base.py

```diff
@@ -16,6 +16,11 @@
 from UltiSnips.text_objects import SnippetInstance
 from UltiSnips.text_objects.python_code import SnippetUtilForAction
 
+# We'll end up compiling the global snippets for every snippet so
+# caching compile() shoud pay off
+from functools import lru_cache
+compile = lru_cache(compile)
```

Comment:
Please do not alias. Instead make an wrapper function: `lru_cached_compile` so the code becomes unambiguous.